rust/src/detect/transforms/xor.rs

```diff
@@ -101,6 +101,18 @@ unsafe extern "C" fn xor_transform(_det: *mut c_void, buffer: *mut c_void, ctx:
     InspectionBufferTruncate(buffer, input_len);
 }
 
+unsafe extern "C" fn xor_serialize(
+    data: *mut *const u8, length: *mut u32, ctx: *mut *const c_void,
+) {
+    if data.is_null() || length.is_null() || ctx.is_null() {
+        return;
+    }
+
+    let ctx = cast_pointer!(ctx, DetectTransformXorData);
+    *data = ctx.key.as_ptr();
+    *length = ctx.key.len() as u32;
```

Comment:
I'm lacking some context with respect to this feature, but nothing is being serialized here.  We're just getting the pointer and length?